embedded/sql/stmt.go

```diff
@@ -91,8 +96,43 @@ const (
 	RightJoin
 )
 
+type TxSummary struct {
+	db *Database
+
+	ces []*store.KV
+	des []*store.KV
+
+	updatedRows     int
+	lastInsertedPKs map[string]uint64
+}
+
+func newTxSummary() *TxSummary {
+	return &TxSummary{
+		lastInsertedPKs: make(map[string]uint64),
+	}
+}
+
+func (s *TxSummary) add(summary *TxSummary) error {
+	if summary == nil {
+		return ErrIllegalArguments
+	}
+
+	s.db = summary.db
```

Comment:
What if databases differ? Cant it even happen? Maybe better to test if databases are the same instead?

Reply:
summary will hold a reference to the latest used database.
The implicit database is set with "UseDatabaseStmt" currently disabled as SQL statement